### DISCUSSION TITLE
Pie Menu for Selection Tools

### DISCUSSION THREAD

**Discussion by JohjohDodo** [wojtryb/Shortcut-Composer#58] [Ideas]
I'm not sure if this is the right place for this, as I'm not experiencing an issue. This is just a suggestion and something I'd really, really, love to see. I use selection tools all the time for how I draw, and having a pie menu with all the selection tools available in one of your pie menus would be amazing (and maybe even an entry for clearing and hiding/showing current selection).

**Comment by wojtryb** [comment]
Hi,

on github you can switch from "main" to "feature/gui-action-editing", then download the .zip again, and import it from krita. It adds new tab in `Configure Shortcut Composer` dialog where you can configure existing actions.

If you want to create your own actions you can:
1. Download VsCode https://code.visualstudio.com/
2. In krita go to: `Settings > Manage Resources > Open Resource Folder`
4. Go to `./pykrita/` and drag&drop `shortcut-composer` onto VsCode
5. Open `actions.py` and `actions.action` files. Let VsCode install the python extension.

You can copy paste any of the multiple assignment actions and fill it with your names and values. VsCode will give you auto-completion and display my hints when you hover over parts of code. Remember to edit both files (definition and implementation).

Creating these from inside krita may come at some point, but it's not the highest priority right now.

**Comment by JohjohDodo** [comment]
Thanks! I'm able to modify an existing menu and turn it into a selection menu with these instructions, but do you have any tips for how I might include a entry in the pie menu for clearing, inverting, and hiding/showing the current selection?

Looking through your code, I'm not quite sure if this is possible or not right now.
I'm guessing I would just include an entry like Toggle.SHOW_SELECTION, Toggle.INVERT_SELECTION, etc. but these functions don't yet exist in core controllers.

My programming skills are pathetic, so I'm not sure if I'm right or not.
Regardless, thank you for your efforts in creating and maintaining this plugin, it really is wonderful.

**Comment by wojtryb** [comment]
Yes, you're correct. Running krita actions like this is not supported. I tried to create a distinction between toggles (actions with on/off state, and usual actions that can only be triggered, and don't have a state)

The latter ones would require a separate controller, but I didn't need it so far. They can now only be triggered with something like Krita.trigger_action(name: str) - so I'm still using strings instead of Enums.

Currently I'm not sure if controller like that is needed (as it would be the only one without a state) and whether it would be useful anywhere else in this plugin.

Tbh I never really needed these as a separate shortcut because RMB when in selection tool work well for me.
